### PR TITLE
migrate (optional) ext-mongo to mongodb/mongodb + ext-mongodb

### PR DESCRIPTION
**Pull request by daniel-sc**
...and drop php 5.3 support

### CODE REVIEW

**Comment by AidasK** [comment]
looking good so far

**Comment by daniel-sc** [comment]
I'll do some tests in the next weeks and remove the draft state when it all works out. 

@AidasK could you check if the Travis build is still green?

**Comment by AidasK** [comment]
not sure if it's working anymore. We will have to rely on manual tests I guess.

**Comment by daniel-sc** [comment]
@AidasK I did some e2e tests and corresponding fixes - for me this is complete - if you have anything to add please let me know!

File: composer.json [hunk 1]
```diff
@@ -19,13 +19,18 @@
...
+      "php": ">=5.4"
   },
   "require-dev": {
     "mikey179/vfsstream": "v1.2.0",
     "league/phpunit-coverage-listener": "~1.1",
     "fabpot/php-cs-fixer": "~2.2",
-    "phpunit/phpunit": "4.*"
+    "phpunit/phpunit": "4.*",
+    "mongodb/mongodb": "^1.4.0",
+    "ext-mongodb": "*"
```
Comment (AidasK) [inline composer.json:30]: maybe add these to suggestions instead? They are optional

**Review by AidasK — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (daniel-sc) [reply composer.json:30]: I added them to "require-dev" as they would be needed to run tests in this repo and consumers do not install these dependencies (only "require"). Furthermore I additionally added "mongodb/mongodb" under "suggest" as - like you noted - this is optional from a consumer perspective. 

Does this make sense?

**Comment by AidasK** [comment]
https://github.com/flowjs/flow-php-server/releases/tag/v1.2.0

**Comment by borakayalar** [comment]
@AidasK is there full mongo implementation, I could not understand mongo/readme 
how should I create "$yourGridFs" ?

**Pull request merged**